cpp/src/utilities/cuda_memcpy.cu

```diff
@@ -26,15 +28,21 @@ namespace cudf::detail {
 
 namespace {
 
+__global__ void copy_kernel(char const* src, char* dst, size_t n)
+{
+  auto const idx = static_cast<size_t>(blockIdx.x) * blockDim.x + threadIdx.x;
```

Comment:
```suggestion
  auto const idx = cudf::detail::grid_1d::global_thread_id();
```

Reply:
Remember to merge the latest commit from 24.08 ;)